src/components/ProfileUserAvatar/ProfileUserAvatar.js

```diff
@@ -0,0 +1,45 @@
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import Image from 'next/image'
+import { useState, useCallback } from 'react'
+
+import useSelectorWithProps from '~/hooks/useSelectorWithProps'
+import { selectAvatarByUserId, withCurrentUserId } from '~/store/selectors'
+
+import UploadAvatarModal from '../UploadAvatarModal'
+import styles from './ProfileUserAvatar.module.scss'
+
+function ProfileUserAvatar () {
+  const userAvatar = useSelectorWithProps({ size: 170 }, withCurrentUserId(selectAvatarByUserId))
+
+  const [showUploadAvatar, setShowUploadAvatar] = useState(false)
+
+  const handleToggleUploadAvatar = useCallback(() => {
+    setShowUploadAvatar(!showUploadAvatar)
+  }, [showUploadAvatar])
```

Comment:
`useState()` setters can accept a function which accepts the current value of the hook. Doing this lets us remove the `useCallback()` dependency on the `useState()` value here. 

```suggestion
  const handleToggleUploadAvatar = useCallback(() => {
    setShowUploadAvatar((prevState) => {
      return !prevState
    })
  }, [])
```